src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -62,21 +62,31 @@ internal override bool IsInsertionTrigger(SourceText text, int insertedCharacter
             completionContext.SuggestionModeItem = CommonCompletionItem.Create(CSharpFeaturesResources.Name, CompletionItemRules.Default);
         }
 
-        private ImmutableArray<ImmutableArray<string>> GetBaseNames(SemanticModel semanticModel,  NameDeclarationInfo nameInfo)
+        private ImmutableArray<ImmutableArray<string>> GetBaseNames(SemanticModel semanticModel, NameDeclarationInfo info)
         {
-            if (nameInfo.Alias != null)
+            ImmutableArray<ImmutableArray<string>> baseNames = default;
+            if (info.Aliases != null)
             {
-                return NameGenerator.GetBaseNames(nameInfo.Alias);
+                baseNames = info.Aliases.ToSet().SelectMany(alias => NameGenerator.GetBaseNames(alias)).ToImmutableArray();
```

Comment:
Can the different aliases generate duplicates?  Should this go into a set/Distinct again?